MCU_Aurix/bootloader/src/isotp.c

```diff
@@ -205,18 +236,22 @@ uint8_t* isotp_rcv(uint32_t* total_length){
  */
 void process_can_to_isotp(uint32_t* rxData, IfxCan_DataLengthCode dlc){
 
-    //TODO: Discuss with sebastian that we would like to set the dlc for all messages.
+    if(dlc <= 0) // Need to assume that the data is also empty, using FBL_NEGATIVE_RESPONSE instead
+        uds_neg_response(FBL_NEGATIVE_RESPONSE, FBL_RC_INCORRECT_MSG_LEN_OR_INV_FORMAT);
 
     if(MAX_ISOTP_MESSAGE_LEN - (iso_RX->write_ptr - iso_RX->data) < dlc || iso_RX->ready_to_read != 0){
 
-        // TODO: send error message that we cannot receive further can messages.
+        // Request with negative response if case of error
+        if(rx_is_single_Frame(rxData, dlc, MAX_FRAME_LEN_CAN))
+            uds_neg_response(rxData[1], FBL_RC_BUSY_REPEAT_REQUEST);
+        else // Using FBL_NEGATIVE_RESPONSE since it is any other frame (Consecutive or Flow Control)
+            uds_neg_response(FBL_NEGATIVE_RESPONSE, FBL_RC_BUSY_REPEAT_REQUEST);
 
         return;
     }
 
     uint8_t* data_ptr = (uint8_t*)rxData;
 
-
     //TODO: Check for flow control and handle  (in case we send data)
     // Single Frame get length of whole isoTP message
     if(((0xF0 & rxData[0]) >> 4) == 0){
```

Comment:
Gehoert jetzt nicht unbedingt zu deinem refactoring aber hier und in den naechsten Zeilen sind mit dem 0xf0, 4, 3 usw. ganz viele magic numbers.